text_mining.py

```diff
@@ -0,0 +1,217 @@
+            word_counts[word] = word_counts.get(word, 0) + 1
+
+    return word_counts
+
+
+def get_top_n_words(word_list, n):
+    """Takes a list of words as input and returns a list of the n most frequently
+    occurring words ordered from most to least frequently occurring.
+
+    Args:
+        word_list: a list of words (assumed to all be in lower case with no punctuation)
+        n: the number of words to return
+
+    Returns:
+        a list of n most frequently occurring words ordered from most to least frequently occurring
+    """
+    word_counts = get_histogram(word_list)
+
+    ordered_by_frequency = sorted(word_counts, key=word_counts.get, reverse=True)
+
+    return ordered_by_frequency[0:n]
+
+
+def sentiment_analyzer(text):
+    """Takes a string of text as input and returns the sentiment analysis of the text.
+    Converts text to string if text is not already a string.
+
+    Args:
+        text: a string of text to be analyzed
+
+    Returns:
+        a sentiment analysis of the text
+    """
+    if type(text) != str:
+        text = ' '.join(text)
+    analyzer = SentimentIntensityAnalyzer()
+    return analyzer.polarity_scores(text)
+
+
+def word_cloud(text, title):
+    """Takes a string of text and a title as input and saves the generated wordcloud as a png
+    with the title as the file name. Converts text to string if text is not already a string.
+
+    Args:
+        text: a string of text used to generate a wordcloud
+        title: the file name of the generated wordcloud
+
+    Returns:
+        nothing; saves the wordcloud to title.png
+    """
+    if type(text) == list:
+        text = ' '.join(text)
+    wordcloud = WordCloud(width = 1000, height = 500, background_color="white").generate(text)
+    wordcloud.to_file('%s_wf.png' % title)
+
+
+if __name__ == "__main__":
+    titles = ('Frankenstein',
+                'Paradise_Lost',
+                'The_Romance_of_Lust',
+                )
+
+    url = {'Frankenstein': 'http://www.gutenberg.org/cache/epub/84/pg84.txt',
+            'Paradise_Lost': 'http://www.gutenberg.org/cache/epub/20/pg20.txt',
+            'The_Romance_of_Lust': 'http://www.gutenberg.org/cache/epub/30254/pg30254.txt',
+            }
+
+    # for each title
+    for title in titles:
+        # get the text from url and strip the header comments
+        text = filter_PG_text(get_cache(url['%s' % title], '%s.txt' % title))
+        # get a list of the filtered words from the text
+        word_list = get_word_list(text)
+        # top n words
+        n = 50
+        # get a list of the top n words
+        top_n_words = (get_top_n_words(word_list, n))
+        # print the top n words
+        print('Top %d Words in %s:' % (n, title))
+        print(top_n_words, '\n')
+        # print the sentiment of the top n words
+        print('Sentiment of Top %d Words in %s:' % (n, title))
+        print(sentiment_analyzer(top_n_words), '\n')
+        # print the sentiment of the whole text
```

Comment:
This might be a little too much commenting - print statements like these mostly stand on their own.

Though, a bit too much documentation is better than a bit too little!